File: BioMetricAuthJava/app/src/main/java/com/example/biometricauthjava/SetPinActivity.java
package com.example.biometricauthjava;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class SetPinActivity extends AppCompatActivity {
    private EditText editPin;
    private Button btnSave;
    private PinStorageManager pinStorage;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_set_pin);

        editPin = findViewById(R.id.editPin);
        btnSave = findViewById(R.id.btnSave);
        pinStorage = new PinStorageManager(this);

        btnSave.setOnClickListener(view -> {
            String pin = editPin.getText().toString();
            if (!pin.isEmpty()) {
                pinStorage.savePin(pin);
                Toast.makeText(this, "PIN saved", Toast.LENGTH_SHORT).show();
                startActivity(new Intent(SetPinActivity.this, HomeActivity.class));
                finish();
            } else {
                Toast.makeText(this, "Please enter a PIN", Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        startActivity(new Intent(this, MainActivity.class));
        finish();
    }
}

File: BioMetricAuthKotlin/app/src/main/java/com/example/biometricauthkotlin/SetPinActivity.kt
package com.example.biometricauthkotlin

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.biometricauthkotlin.databinding.ActivitySetPinBinding
import kotlin.jvm.java

class SetPinActivity : AppCompatActivity() {

    private lateinit var binding : ActivitySetPinBinding
    private lateinit var pinStorage: PinStorageManager


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivitySetPinBinding.inflate(layoutInflater)
        setContentView(binding.root)

        pinStorage = PinStorageManager(this)

        binding.btnSave.setOnClickListener {
            val pin = binding.editPin.text.toString()
            if (pin.isNotEmpty()) {
                pinStorage.savePin(pin)
                Toast.makeText(this, "PIN saved successfully", Toast.LENGTH_SHORT).show()
                startActivity(Intent(this, HomeActivity::class.java))
                finish()
            } else {
                Toast.makeText(this, "Please enter a PIN", Toast.LENGTH_SHORT).show()
            }
        }

    }

    override fun onBackPressed() {
        super.onBackPressed()
        startActivity(Intent(this, MainActivity::class.java))
        finish()
    }
}